force-app/main/default/lwc/propertyLocation/propertyLocation.js

```diff
@@ -0,0 +1,97 @@
+import { LightningElement, wire, api } from 'lwc';
+import { getLocationService } from 'lightning/mobileCapabilities';
+import { getRecord, getFieldValue } from 'lightning/uiRecordApi';
+
+// Using hardcoded fields due to LWC bug
+const LATITUDE_FIELD = 'Property__c.Location__Latitude__s';
+const LONGITUDE_FIELD = 'Property__c.Location__Longitude__s';
+
+const fields = [LATITUDE_FIELD, LONGITUDE_FIELD];
+
+export default class PropertyLocation extends LightningElement {
+    error;
+    myDeviceLocationService;
+    myLocation;
+    @api recordId;
+
+    @wire(getRecord, { recordId: '$recordId', fields })
+    wiredProperty({ data, error }) {
+        if (data) {
+            this.property = data;
+            this.error = undefined;
+        } else if (error) {
+            this.error = error;
+            this.property = undefined;
+        }
+    }
+
+    connectedCallback() {
+        this.myDeviceLocationService = getLocationService();
+        if (this.myDeviceLocationService.isAvailable()) {
+            // Running on the Salesforce mobile app on a device
+            this.getLocationFromMobileDevice();
+        } else if (navigator.geolocation) {
+            // Running on a browser
+            this.getLocationFromBrowser();
+        } else {
+            this.error = 'No location services available';
+        }
+    }
+
+    getLocationFromMobileDevice() {
+        this.myDeviceLocationService
+            .getCurrentPosition()
+            .then((result) => {
+                this.myLocation = result.coords;
+                this.calculateDistance();
+            })
+            .catch((error) => {
+                this.error = error;
+            });
+    }
+
+    getLocationFromBrowser() {
+        navigator.geolocation.getCurrentPosition(
+            (result) => {
+                this.myLocation = result.coords;
+                this.calculateDistance();
+            },
+            (error) => {
+                this.error = error;
+            }
+        );
+    }
+
+    get distance() {
```

Comment:
I think that it's great to use a getter to call `calculateDistance()`.
Why do you need to call it also in `getLocationFromMobileDevice()` and `getLocationFromBrowser()`? The getter should automatically recalculate the distance when `myLocation` changes in either of these two methods.

Reply:
Because I missed that when refactoring :) Thanks!